### PR TITLE
Make compiler compatible with later versions of react-with-styles

### PR DESCRIPTION
**Pull request by majapw**
This change makes it so that the CSS compiler is compatible with react-with-styles@>2.2.0. Previously, because StyleSheet creation was done on import, all we had to do was walk the require tree. Now we actually need to render the components (although we don't really care about the contents).

One thing that this will require is that... when building CSS for a library, ReactDOM.render will need to be called (which was not previously true). 

to: @lencioni @ljharb @airbnb/webinfra

### CODE REVIEW

File: packages/compiler/src/utils.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import ReactDOMServer from 'react-dom/server';
```
Comment (ljharb) [inline packages/compiler/src/utils.js]: Is this present in 0.14?

Comment (majapw) [reply packages/compiler/src/utils.js]: I mean 0.14 was before the split, yeah? 

I think the server existed but possibly with a different import.

Comment (ljharb) [reply packages/compiler/src/utils.js]: Right; i think it’s ReactDOM.render there. We may need a file that does conditional try catch requires to get at the proper render method.

Comment (majapw) [reply packages/compiler/src/utils.js]: Wait, this totally existed in React 0.14 --- https://reactjs.org/blog/2015/10/07/react-v0.14.html

Comment (majapw) [reply packages/compiler/src/utils.js]:
```
Two Packages: React and React DOM
As we look at packages like react-native, react-art, react-canvas, and react-three, it has become clear that the beauty and essence of React has nothing to do with browsers or the DOM.

To make this more clear and to make it easier to build more environments that React can render to, we’re splitting the main react package into two: react and react-dom. This paves the way to writing components that can be shared between the web version of React and React Native. We don’t expect all the code in an app to be shared, but we want to be able to share the components that do behave the same across platforms.

The react package contains React.createElement, .createClass, .Component, .PropTypes, .Children, and the other helpers related to elements and component classes. We think of these as the isomorphic or universal helpers that you need to build components.

The react-dom package has ReactDOM.render, .unmountComponentAtNode, and .findDOMNode. In react-dom/server we have server-side rendering support with ReactDOMServer.renderToString and .renderToStaticMarkup.
```

**Pull request merged**